src/asmcnc/skavaUI/screen_job_recovery.py

```diff
@@ -329,16 +331,56 @@ def on_pre_enter(self):
             self.display_list[self.selected_line_index + 6] = "[color=FF0000]" + self.display_list[self.selected_line_index + 6] + "[/color]"
             self.update_display()
 
-    def scroll_up(self):
+    def start_scrolling_up(self):
+        self.scrolling_up = True
+
+        # Do one scroll immediately
+        self.do_scroll_up()
+
+        # Start scrolling after a delay
+        self.scroll_up_event = Clock.schedule_once(self.scroll_up, 0.5)
+
+    def do_scroll_up(self):
         if self.selected_line_index > 0:
             self.selected_line_index -= 1
             self.update_display()
 
-    def scroll_down(self):
+    def scroll_up(self, dt=0):
+        if self.scrolling_up:
+            self.do_scroll_up()
+
+            # Keep scrolling until button released
+            self.scroll_up_event = Clock.schedule_once(self.scroll_up, 0.03)
```

Comment:
I recommend declaring scroll_up_event and scroll_down_event at the top of the class as: 

scroll_up_event = None
scroll_down_event = None

just in case there's a freak situation where the stop function unschedules the event before it's been created - this is rare but I have seen it happen in other cases, and it will cause a crash. 

Then when unscheduling these events, do:

if event != None: Clock.unschedule(event)